python/ray/ml/trainer.py

```diff
@@ -54,6 +58,21 @@ def __init__(
 
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def training_function(
+        self, config: Dict[str, Any], checkpoint: Optional[Checkpoint] = None
+    ):
+        """Function to define the training logic.
+
+        `self.train_dataset` and the Dataset values in `self.extra_datasets`
+        have already been preprocessed by `self.preprocessor`.
+
+        Args:
+            config: Configurations for training logic specific implementation.
+            checkpoint: A checkpoint to resume from, if applicable.
+        """
```

Comment:
What is the return value?

Reply:
There is none

Reply:
Let's document that (and that results are returned via tune)